src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -53,7 +55,12 @@ const CosmosD16s = "Cosmos.D16s";
 
 const onSKUChange = (newValue: InputType, currentValues: Map<string, SmartUiInput>): Map<string, SmartUiInput> => {
   currentValues.set("sku", { value: newValue });
-  currentValues.set("costPerHour", { value: costPerHourValue });
+
+  // Update cost per hour based on new sku
+  currentValues.set("costPerHour", {
+    value: calculateCost(newValue as string, currentValues.get("NumberOfInstances").value as number),
```

Comment:
this should be "instances" right?
This scenario is broken :
Account does not have sqlx provisioned, go to Dedicated gateway blade, change the Sku type.